src/quo/components/share/share_qr_code/component_spec.cljs

```diff
@@ -1,34 +1,119 @@
-               {:link-title " A test title"}])
-    (-> (js/expect (h/get-by-text "A test title"))
-        (.toBeTruthy)))
-
-  (h/test "renders share qr code url"
-    (h/render [share-qr-code/view
-               {:qr-url " A test url"}])
-    (-> (js/expect (h/get-by-text "A test url"))
-        (.toBeTruthy)))
-
-  (h/test "on press link event fires"
-    (let [event (h/mock-fn)]
-      (h/render [share-qr-code/view
-                 {:url-on-press event
-                  :qr-url       " A test url"}])
-      (h/fire-event :press (h/get-by-text "A test url"))
-      (-> (js/expect event)
-          (.toHaveBeenCalledTimes 1))))
-
-  (h/test "on press share event fires"
-    (let [event (h/mock-fn)]
-      (h/render [share-qr-code/view
-                 {:share-on-press event}])
-      (h/fire-event :press (h/get-by-label-text :share-profile))
-      (-> (js/expect event)
-          (.toHaveBeenCalledTimes 1)))))
+  (h/describe "Renders share-qr-code component in all types"
+    (let [qr-label "Text shown below QR"]
+      (h/test "Profile"
+        (render-share-qr-code {:type    :profile
+                               :qr-data qr-label})
+        (h/is-truthy (h/get-by-text qr-label)))
+
+      (h/test "Wallet Legacy"
+        (render-share-qr-code {:type    :wallet-legacy
+                               :qr-data qr-label
+                               :emoji   "👻"})
+        (h/is-truthy (h/get-by-text qr-label)))
+
+      (h/test "Wallet Multichain"
+        (render-share-qr-code {:type    :wallet-multichain
+                               :qr-data qr-label
+                               :emoji   "👻"})
+        (h/is-truthy (h/get-by-text qr-label)))))
+
+  (h/describe "Fires all events for all types"
+    (letfn [(test-fire-events [props test-seq]
+              (doseq [{:keys [test-name event-name
+                              callback-prop-key
+                              accessibility-label]} test-seq
+                      :let                          [event-fn (h/mock-fn)]]
+                (h/test test-name
+                  (render-share-qr-code (assoc props callback-prop-key event-fn))
+                  (h/fire-event event-name (h/get-by-label-text accessibility-label))
+                  (h/was-called-times event-fn 1))))]
+
+      (h/describe "Profile"
+        (test-fire-events
+         {:type :profile}
+         [{:test-name           "Text pressed"
+           :accessibility-label :share-qr-code-info-text
+           :event-name          :press
+           :callback-prop-key   :on-text-press}
+          {:test-name           "Text long pressed"
+           :accessibility-label :share-qr-code-info-text
+           :event-name          :long-press
+           :callback-prop-key   :on-text-long-press}
+          {:test-name           "Share button"
+           :accessibility-label :link-to-profile
+           :event-name          :press
+           :callback-prop-key   :on-share-press}]))
```

Comment:
Seriously thinking about creating this as a macro instead of being a function, it reduced the code complexity a lot and we just need to describe the test using a map :thinking: 
WDYT